codegen/utils.ts

```diff
@@ -0,0 +1,117 @@
+import * as M from "../frame_metadata/mod.ts";
+
+export const importSource = new URL("../mod.ts", import.meta.url).toString();
+
+export type File = { getContent: () => S };
+export type Files = Map<string, File>;
+
+export type S = string | number | S[];
+
+export namespace S {
+  export function array(items: S[]): S {
+    return ["[", items.map((x) => [x, ","]), "]"];
+  }
+  export function object(
+    ...items: (readonly [doc: S, prop: S, val: S] | readonly [prop: S, val: S])[]
+  ): S {
+    return ["{", items.map((x) => [x.slice(0, -1), ":", x.at(-1)!, ","]), "}"];
+  }
+  export function string(value: string): S {
+    return JSON.stringify(value);
+  }
+  export function toString(value: S): string {
+    if (!(value instanceof Array)) return value.toString();
+    const parts = value.map(S.toString);
+    return parts.map((x) => x.trim()).join(parts.some((x) => x.includes("\n")) ? "\n" : " ").trim();
+  }
+}
+
+export type Decl = { path: string; code: S };
+
+export function getPath(tys: M.Ty[], ty: M.Ty): string | null {
+  if (ty.type === "Struct" && ty.fields.length === 1 && ty.params.length) return null;
+  return _getName(ty);
+
+  function _getName(ty: M.Ty): string | null {
+    if (ty.type === "Primitive") {
+      return ty.kind;
+    }
+    if (ty.type === "Compact") {
+      return null;
+    }
+    if (ty.path.at(-1) === "Era") return "Era";
```

Comment:
Might a pallet developer expose a different type with the name `Era`?

Reply:
This is also an issue with the codec derivation. #259